inlang/source-code/github-lint-action/src/main.ts

```diff
@@ -1,97 +1,388 @@
-		  `,
+			issue_number: prNumber as number,
+			body: commentContent,
+			as: "ninja-i18n",
 		})
 	} catch (error) {
 		// Fail the workflow run if an error occurs
-		if (error instanceof Error) core.setFailed(error.message)
+		if (error instanceof Error) core.setFailed(error)
 	}
 }
+
+export default run
+
+function createLintSummary(
+	reportsHead: MessageLintReport[],
+	reportsBase: MessageLintReport[],
+	installedRules: InstalledMessageLintRule[]
+) {
+	const summary: { id: string; name: string; count: number }[] = []
+	const diffReports = reportsHead.filter(
+		(report) =>
+			!reportsBase.some(
+				(baseReport) =>
+					baseReport.ruleId === report.ruleId &&
+					baseReport.languageTag === report.languageTag &&
+					baseReport.messageId === report.messageId
+			)
+	)
+	for (const installedRule of installedRules) {
+		const id = installedRule.id
+		const name =
+			typeof installedRule.displayName === "object"
+				? installedRule.displayName.en
+				: installedRule.displayName
+		const count = diffReports.filter((report) => report.ruleId === id).length
+		if (count > 0) {
+			summary.push({ id, name, count: count })
+		}
+	}
+	const changedIds = diffReports
+		.map((report) => report.messageId)
+		.filter((value, index, self) => self.indexOf(value) === index)
+
+	return { summary, changedIds }
+}
+
+// All functions below will be replaced by the @lix-js/client package in the future
+
+// Function to checkout a branch
+async function checkoutBranch(branchName: string) {
+	return new Promise<void>((resolve, reject) => {
+		// Execute the git command to checkout the branch
+		exec(`git checkout ${branchName}`, { cwd: process.cwd() }, (error, stdout, stderr) => {
+			if (error) {
+				console.error(`Error executing command: ${error}`)
+				reject(error)
+				return
+			}
+			core.debug(`stdout: ${stdout}`)
+			core.debug(`stderr: ${stderr}`)
+			resolve()
+		})
+	})
+}
+
+// Function to fetch the branches
+async function fetchBranch(branchName: string) {
+	return new Promise<void>((resolve, reject) => {
+		// Execute the git command to fetch the branch
+		exec(`git fetch origin ${branchName}`, { cwd: process.cwd() }, (error, stdout, stderr) => {
+			if (error) {
+				console.error(`Error executing command: ${error}`)
+				reject(error)
+				return
+			}
+			core.debug(`stdout: ${stdout}`)
+			core.debug(`stderr: ${stderr}`)
+			resolve()
+		})
+	})
+}
+
+// Funtion to pull latest changes
+async function pull() {
+	return new Promise<void>((resolve, reject) => {
+		// Execute the git command to pull the latest changes
+		exec(`git pull`, { cwd: process.cwd() }, (error, stdout, stderr) => {
+			if (error) {
+				console.error(`Error executing command: ${error}`)
+				reject(error)
+				return
+			}
+			core.debug(`stdout: ${stdout}`)
+			core.debug(`stderr: ${stderr}`)
+			resolve()
+		})
+	})
+}
+
+// Function to clone the base repository
+async function cloneRepository(repoData: { link: string; branch: string }) {
```

Comment:
not sure why needed, this is allready in the yaml as one of the steps before this code runs, no?